code/game/objects/items/RSF.dm

```diff
@@ -3,183 +3,203 @@ CONTAINS:
+								/obj/item/paper = 10,
+								/obj/item/storage/pill_bottle/dice = 200,
+								/obj/item/pen = 50,
+								/obj/item/clothing/mask/cigarette = 10,
+								)
+	///An associated list of fuel and it's value
+	var/list/matter_by_item = list(/obj/item/rcd_ammo = 10,)
+	///A list of surfaces that we are allowed to place things on.
+	var/list/allowed_surfaces = list(/turf/open/floor, /obj/structure/table)
+	///The unit of mesure of the matter, for use in text
+	var/discriptor = "fabrication-units"
+	///The verb that describes what we're doing, for use in text
+	var/action_type = "Dispensing"
+
+/obj/item/rsf/Initialize()
+	. = ..()
+	to_dispense = cost_by_item[1]
+	dispense_cost = cost_by_item[to_dispense]
 
 /obj/item/rsf/examine(mob/user)
 	. = ..()
-	. += "<span class='notice'>It currently holds [matter]/30 fabrication-units.</span>"
+	. += "<span class='notice'>It currently holds [matter]/[max_matter] [discriptor].</span>"
 
 /obj/item/rsf/cyborg
 	matter = 30
 
 /obj/item/rsf/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/rcd_ammo))
-		if((matter + 10) > 30)
-			to_chat(user, "The RSF can't hold any more matter.")
+	if(is_type_in_list(W,matter_by_item))//If the thing we got hit by is in our matter list
+		var/tempMatter = matter_by_item[W.type] + matter
+		if(tempMatter > max_matter)
+			to_chat(user, "<span class='warning'>\The [src] can't hold any more [discriptor]!</span>")
 			return
 		qdel(W)
-		matter += 10
+		matter = tempMatter //We add its value
 		playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
-		to_chat(user, "The RSF now holds [matter]/30 fabrication-units.")
+		to_chat(user, "<span class='notice'>\The [src] now holds [matter]/[max_matter] [discriptor].</span>")
+		icon_state = initial(icon_state)//and set the icon state to the initial value it had
 	else
 		return ..()
 
 /obj/item/rsf/attack_self(mob/user)
 	playsound(src.loc, 'sound/effects/pop.ogg', 50, 0)
-	switch(mode)
-		if(5)
-			mode = 1
-			to_chat(user, "Changed dispensing mode to 'Drinking Glass'")
-		if(1)
-			mode = 2
-			to_chat(user, "Changed dispensing mode to 'Paper'")
-		if(2)
-			mode = 3
-			to_chat(user, "Changed dispensing mode to 'Pen'")
-		if(3)
-			mode = 4
-			to_chat(user, "Changed dispensing mode to 'Dice Pack'")
-		if(4)
-			mode = 5
-			to_chat(user, "Changed dispensing mode to 'Cigarette'")
+	var/target = cost_by_item
+	var/cost = 0
+	//Warning, prepare for bodgecode
+	while(islist(target))//While target is a list we continue the loop
+		var/picked = show_radial_menu(user, src, formRadial(target), custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE)
+		if(!check_menu(user) || picked == null)
+			return
+		for(var/emem in target)//Back through target agian
+			var/atom/test = OBJECT_OR_LIST_ELEMENT(target, emem)
+			if(picked == initial(test.name))//We try and find the entry that matches the radial selection
+				cost = target[emem]//We cash the cost
+				target = emem
```

Comment:
How come TG does their comments wacky but then does it our way here I cry

Reply:
Because this isn't autodocced, this is a regular comment.